v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapper.java

```diff
@@ -277,6 +330,49 @@ private static PTransform<PBegin, PCollection<SourceRow>> getJdbcIO(
     return jdbcIO;
   }
 
+  /**
+   * Private helper to construct {@link ReadWithUniformPartitions} as per the reader configuration.
+   *
+   * @param config Configuration.
+   * @param dataSourceConfiguration dataSourceConfiguration (which is derived earlier from the
+   *     reader configuration)
+   * @param tableConfig discovered table configurations.
+   * @param sourceTableSchema schema of the source table.
+   * @return
+   */
+  private static PTransform<PBegin, PCollection<SourceRow>> getReadWithUniformPartitionIO(
+      JdbcIOWrapperConfig config,
+      DataSourceConfiguration dataSourceConfiguration,
+      SourceSchemaReference sourceSchemaReference,
+      TableConfig tableConfig,
+      SourceTableSchema sourceTableSchema) {
+
+    ReadWithUniformPartitions.Builder<SourceRow> readWithUniformPartitionsBuilder =
+        ReadWithUniformPartitions.<SourceRow>builder()
+            .setTableName(tableConfig.tableName())
+            .setPartitionColumns(tableConfig.partitionColumns())
+            .setDataSourceProviderFn(JdbcIO.PoolableDataSourceProvider.of(dataSourceConfiguration))
+            .setDbAdapter(new MysqlDialectAdapter(MySqlVersion.DEFAULT))
+            .setApproxTotalRowCount(tableConfig.approxRowCount())
+            .setRowMapper(
+                new JdbcSourceRowMapper(
+                    config.valueMappingsProvider(),
+                    sourceSchemaReference,
+                    sourceTableSchema,
+                    config.shardID()))
+            .setWaitOnSignals(config.waitOnSignals())
+            // TODO(vardhanvthigle): if index is not of the type of a single auto incrementing key,
+            // don't set this.
+            .setSplitStageCountHint(4L)
+            .setRangesPeek(config.rangesPeek());
```

Comment:
nit: Method name does not clarify what it does. maybe call it - additionalOperationsOnRange or something less verbose.

Reply:
makes sense. I borrowed the `peek` suffix from the java stream api. It was chosen to underscore that ptransform is meant to be a consumer like transform.

Will change `additionalOperationsOnRange